grammar.js

```diff
@@ -743,7 +743,10 @@ module.exports = grammar({
     )),
 
     pipe_expression: $ => prec.left(seq(
-      $.primary_expression,
+      choice(
+        $.primary_expression,
+        $.block,
+      ),
       choice('->', '|>'),
       choice(
         $.value_identifier,
```

Comment:
We can remove all rules on rhs and use only `choice($.primary_expression, $.block`)?

There will be minor changes in testing. But I think it's more consistent.

Example:

```res
1 -> #A("foo")
```

Is parsed as:

```
yarn run v1.22.19
$ /home/pedro/Desktop/projects/tree-sitter-rescript/node_modules/.bin/tree-sitter parse ../../test-filetypes/rescript_print.res
(source_file [0, 0] - [1, 0]
  (expression_statement [0, 0] - [0, 14]
    (call_expression [0, 0] - [0, 14]
      function: (pipe_expression [0, 0] - [0, 7]
        (number [0, 0] - [0, 1])
        (polyvar_identifier [0, 5] - [0, 7]))
      arguments: (arguments [0, 7] - [0, 14]
        (string [0, 8] - [0, 13]
          (string_fragment [0, 9] - [0, 12]))))))
Done in 0.47s.

```

With this change:

```
yarn run v1.22.19
$ /home/pedro/Desktop/projects/tree-sitter-rescript/node_modules/.bin/tree-sitter parse ../../test-filetypes/rescript_print.res
(source_file [0, 0] - [1, 0]
  (expression_statement [0, 0] - [0, 14]
    (pipe_expression [0, 0] - [0, 14]
      (number [0, 0] - [0, 1])
      (polyvar [0, 5] - [0, 14]
        (polyvar_identifier [0, 5] - [0, 7])
        (arguments [0, 7] - [0, 14]
          (string [0, 8] - [0, 13]
            (string_fragment [0, 9] - [0, 12])))))))
Done in 0.05s.

```

Reply:
Actually in another PR

Reply:
Yep. Fair point, thank you for the PR!